tests/cephfs/cephfs_mirroring/test_cephfs_mirror_disconnect.py

```diff
@@ -0,0 +1,461 @@
+import json
+import random
+import string
+import time
+import traceback
+
+from ceph.ceph import CommandFailed
+from ceph.parallel import parallel
+from tests.cephfs.cephfs_mirroring.cephfs_mirroring_utils import CephfsMirroringUtils
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+from utility.retry import retry
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    CEPH-83574100 - Validate all failure scenarios to disconnect the mirroring.
+    It has pre-requisites,
+    performs various operations, and cleans up the system after the tests.
+
+    Scenarios Covered :
+    Sceneario 1 : Disconnect the mirroring Node network for 90 seconds.
+    Sceneario 2 : Disconnect the mirroring by removing the peer connection.
+    Sceneario 3 : Disconnect mds nodes on source cluster
+    Sceneario 4 : Disconnect mds nodes on target cluster
+
+    Args:
+        ceph_cluster: The Ceph cluster to perform the mirroring tests on.
+        **kw: Additional keyword arguments.
+
+    Returns:
+        int: 0 if the test is successful, 1 if there's an error.
+
+    Raises:
+        Exception: Any unexpected exceptions that might occur during the test.
+    """
+
+    try:
+        config = kw.get("config")
+        ceph_cluster_dict = kw.get("ceph_cluster_dict")
+        fs_util_ceph1 = FsUtils(ceph_cluster_dict.get("ceph1"))
+        fs_util_ceph2 = FsUtils(ceph_cluster_dict.get("ceph2"))
+        fs_mirroring_utils = CephfsMirroringUtils(
+            ceph_cluster_dict.get("ceph1"), ceph_cluster_dict.get("ceph2")
+        )
+        build = config.get("build", config.get("rhbuild"))
+        source_clients = ceph_cluster_dict.get("ceph1").get_ceph_objects("client")
+        target_clients = ceph_cluster_dict.get("ceph2").get_ceph_objects("client")
+        cephfs_mirror_node = ceph_cluster_dict.get("ceph1").get_ceph_objects(
+            "cephfs-mirror"
+        )
+
+        log.info("checking Pre-requisites")
+        if not source_clients or not target_clients:
+            log.info(
+                "This test requires a minimum of 1 client node on both ceph1 and ceph2."
+            )
+            return 1
+        log.info("Preparing Clients...")
+        fs_util_ceph1.prepare_clients(source_clients, build)
+        fs_util_ceph2.prepare_clients(target_clients, build)
+        fs_util_ceph1.auth_list(source_clients)
+        fs_util_ceph2.auth_list(target_clients)
+        source_fs = "cephfs_nw_1"
+        target_fs = "cephfs_rem_1"
+        target_user = "mirror_remote_nw"
+        target_site_name = "remote_site_nw"
+        fs_details = fs_util_ceph1.get_fs_info(target_clients[0], source_fs)
```

Comment:
Shouldn't this be the fs details on source clients? source_clients[0] instead of target_clients[0] ?